api/envoy/config/route/v3/route_components.proto

```diff
@@ -2070,6 +2097,9 @@ message RateLimit {
 
       // Rate limit on masked remote address.
       MaskedRemoteAddress masked_remote_address = 10;
+
+      // Rate limit on the existence of query parameters.
+      QueryParameterValueMatch query_parameter_value_match = 11;
```

Comment:
A high-level question before going into this. Why isn't this a `rate_limit_descriptor` extension?
Making it an extension will avoid this from being linked if not needed.

Reply:
@adisuissa thanks! but i am not familiar with `rate_limit_descriptor` extension, can it also support various types of matches(prefix, suffix, regex...) like QueryParameterValueMatch?

Reply:
There's an extension [here](https://github.com/envoyproxy/envoy/blob/a1606b267c27a6ffc057ac3abe7ae38734088265/source/extensions/rate_limit_descriptors/expr/config.h) that introduces a CEL rate-limiter.

That said, if you think there's a good reason for this to be integral part of Envoy's core, LMK.

Reply:
i think `rate_limit_descriptor` extension does not have all the capabilities of `QueryParameterMatcher` which supports different types of match for each query parameter.